src/Workspaces/Core/Portable/Shared/TestHooks/TaskExtensions.cs

```diff
@@ -46,122 +45,5 @@ static Task CompletesTrackingOperationSlow(Task task, IDisposable token)
                     TaskScheduler.Default);
             }
         }
-
-        // Following code is copied from Microsoft.VisualStudio.Threading.TplExtensions (renamed to avoid ambiguity)
-        // https://github.com/microsoft/vs-threading/blob/main/src/Microsoft.VisualStudio.Threading/TplExtensions.cs
```

Comment:
moved this code to shared location since it is needed by code that is also shared.

Reply:
I'm surprised that WeaklyCachedRecoverableValueSource is being used anywhere other than workspaces in the first place -- instead of moving this to the shared utilities, can we move WeaklyCachedRecoverableValueSource in the other direction?

Reply:
Yes.  You're right.  Moved that type to only be in the workspace.  Def nicer.